filecoin-proofs/src/api/internal.rs

```diff
@@ -335,35 +343,43 @@ pub fn generate_post_fixed_sectors_count(
         cs
     };
 
-    let pub_inputs = vdf_post::PublicInputs {
+    let pub_inputs = beacon_post::PublicInputs {
         challenge_seed: PedersenDomain::try_from_bytes(&safe_challenge_seed).unwrap(),
         commitments,
         faults: Vec::new(),
     };
 
-    let trees: Vec<Tree> = fixed
-        .input_parts
-        .iter()
-        .map(|(access, _)| {
-            if let Some(s) = &access {
-                make_merkle_tree(
-                    s,
-                    PaddedBytesAmount(pub_params.vanilla_params.sector_size as u64),
-                )
-                .unwrap()
-            } else {
-                panic!("faults are not yet supported")
-            }
-        })
-        .collect();
+    let sector_size = pub_params.vanilla_params.vdf_post_pub_params.sector_size as u64;
+    let bytes = PaddedBytesAmount(sector_size);
+
+    let mut replicas = Vec::with_capacity(fixed.input_parts.len());
+    let mut trees = Vec::with_capacity(fixed.input_parts.len());
+
+    for (access, _) in &fixed.input_parts {
+        if let Some(sealed_path) = access {
+            let mut f_in = File::open(sealed_path).expect("invalid path");
+            let mut replica = Vec::new();
+            f_in.read_to_end(&mut replica)
+                .expect("failed to read replica");
 
+            trees.push(public_params(bytes).graph.merkle_tree(&replica)?);
+            replicas.push(replica);
+        } else {
+            panic!("faults are not yet supported")
```

Comment:
Now that you're editing this code... will you think of a way to remove this panic?

Reply:
I will work on faults likely tomorrow as a follow up to this and think about handling it then